storybookjs/storybook
# Vite: Add vite framework plugin if not found

Issue: https://github.com/storybookjs/builder-vite/issues/498

Closes https://github.com/storybookjs/storybook/pull/19245

## What I did

When we made the change in 7.0 to start using the user's `vite.config.js`, I made the (faulty) assumption that users would already have vite framework plugins (e.g. `@vitejs/plugin-react`) installed.  But this isn't always true, for example, a project might just be a collection of components being exported, and there may not be an actual app.  

So, this PR checks to see if the required framework plugin is already in the config, and if not, adds it.  I created a utility function to check for the plugin, but ideally this would live in a `vite-core` instead of duplicated across each framework.  This, combined with the duplication in https://github.com/storybookjs/storybook/pull/19216, makes me lean towards creating such a package.  But, it can be done any time and these duplications cleaned up at that point, so I don't think it's a blocker.

## How to test

Create a sandbox, delete `vite.config.js`, and start storybook.  Without this change there would be a crash, but it should work just fine now.
